Fix validation of starts and finals for a2scanct, a3scanct, ...

reszelaz opened the pull request:
Basically the a2scanct, a3scanct does not work now due to a bug in validating the starts and finals positions.

The truth value of an array with more than one element is ambiguous. Use a.all() to determine that all motors have the same start and final position.

rhomspuron commented:
I tested the patch and it breaks the meshct:
```
  File "/homelocal/sicilia/rhoms/pycharm/sardana/src/sardana/macroserver/msmacromanager.py", line 1462, in runMacro
    for step in macro_obj.exec_():
  File "/homelocal/sicilia/rhoms/pycharm/sardana/src/sardana/macroserver/macro.py", line 2315, in exec_
    for i in it:
  File "/homelocal/sicilia/rhoms/pycharm/sardana/src/sardana/macroserver/macros/scan.py", line 1786, in run
    for step in self._gScan.step_scan():
  File "/homelocal/sicilia/rhoms/pycharm/sardana/src/sardana/macroserver/scan/gscan.py", line 918, in step_scan
    for i in self.scan_loop():
  File "/homelocal/sicilia/rhoms/pycharm/sardana/src/sardana/macroserver/scan/gscan.py", line 2376, in scan_loop
    self.go_through_waypoints()
  File "/homelocal/sicilia/rhoms/pycharm/sardana/src/sardana/macroserver/scan/gscan.py", line 1240, in go_through_waypoints
    self._go_through_waypoints()
  File "/homelocal/sicilia/rhoms/pycharm/sardana/src/sardana/macroserver/scan/gscan.py", line 2161, in _go_through_waypoints
    if all(self.macro.starts == self.macro.finals):
TypeError: 'bool' object is not iterable

```

reszelaz commented:
@rhomspuron thanks for the test that you performed. The bug resulted to be in the `meshct` macro. I've fixed it in 5aa7101.

I've also added automatic tests for `a2scanct` and `meshct`. 

I will merge it now as soon as travis-ci allows me.
